packages/xo-web/src/xo-app/backup-ng/new/_schedules/modal.js

```diff
@@ -0,0 +1,100 @@
+import _ from 'intl'
+import decorate from 'apply-decorators'
+import defined from '@xen-orchestra/defined'
+import Icon from 'icon'
+import React from 'react'
+import Scheduler, { SchedulePreview } from 'scheduling'
+import { Card, CardBlock } from 'card'
+import { generateId } from 'reaclette-utils'
+import { injectState, provideState } from 'reaclette'
+import { Number } from 'form'
+
+import { FormGroup, Input } from '../../utils'
+
+import { isRetentionsMissing } from '.'
+
+export default decorate([
+  provideState({
+    effects: {
+      initialize() {
+        this.props.retentions.forEach(({ valuePath }) => {
+          this.effects[`set${valuePath}`] = value => {
```

Comment:
Is this allowed by Reaclette?

Reply:
It's not in the documentation but we can do it.

Reply:
Are you sure about this? How would it behave with inheritance? This seems dangerous.

Reply:
Why it should have a side effect on the inheritance? if i can access to the effect's ref i can add properties without any problems.

Reply:
@julien-f  What do you think about that?

Reply:
It's not supported by Reaclette.

It may work and it may break, you should avoid this.